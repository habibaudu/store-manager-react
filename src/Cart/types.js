const MAKE_SALES = 'MAKE_SALES'
const MAKE_SALES_ERROR = 'MAKE_SALES'

export  default { MAKE_SALES, MAKE_SALES_ERROR }