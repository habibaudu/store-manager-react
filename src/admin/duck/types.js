const LOAD_ALL_PRODUCT = 'LOAD_PRODUCTS';
const ADD_PRODUCT = 'ADD_PRODUCT';
const SET_ADD_STATE = 'SET_ADD_STATE';
const SET_ADD_ERROR = 'SET_ADD_ERROR';
const PRODUCT_SUCESSFULLY_ADDED = 'PRODUCT_SUCESSFULLY_ADDED';
const ADDING_PRODUCT = 'ADDING_PRODUCT';
const ADD_ERROR = 'ADD_ERROR';

export default { LOAD_ALL_PRODUCT, ADD_PRODUCT,PRODUCT_SUCESSFULLY_ADDED,ADDING_PRODUCT,ADD_ERROR,SET_ADD_STATE,SET_ADD_ERROR};
