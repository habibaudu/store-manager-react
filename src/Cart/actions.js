import types from './types';

const setSalesRecord = (salesState) => ({
    type:types.MAKE_SALES,
    salesState
});

const setSalesError = (salesError) => ({
  type:types.MAKE_SALES_ERROR,
  salesError
});




export default { setSalesRecord, setSalesError };
