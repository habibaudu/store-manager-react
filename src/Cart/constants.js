export default {
  MAKING_SALES: 'MAKING_SALES',
  SALES_SUCESS: 'SALES_SUCESS',
  SALES_ERROR: 'SALES_ERROR',
};