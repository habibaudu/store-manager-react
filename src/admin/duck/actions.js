import types from './types';

const setAllProducts = products => ({
  type: types.LOAD_ALL_PRODUCT,
  payload: products,
});

const addProducts = productadded => ({
  type: types.ADD_PRODUCT,
  productadded,
});

const addProductStatus = status => ({
  type: types.SET_ADD_STATE,
  status,
});

const addProductError = errormessage => ({
  type: types.SET_ADD_ERROR,
  errormessage,
});

const setDeleteProduct = id => ({
  type: types.DELETE_PRODUCT,
  id,
});

const setSignupState = signupState => ({
  type: types.SET_SIGNUP_STATE,
  signupState,
});
const setSignupError = errorMessage => ({
  type: types.SET_SIGNUP_ERROR,
  errorMessage,
});

export default {
  setAllProducts,
  addProducts,
  addProductStatus,
  addProductError,
  setDeleteProduct,
  setSignupState,
  setSignupError,
};
