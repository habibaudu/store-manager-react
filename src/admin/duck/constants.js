export default {
  PRODUCT_SUCESSFULLY_ADDED: 'PRODUCT_SUCESSFULLY_ADDED',
  ADDING_PRODUCT: 'ADDING_PRODUCT',
  ADD_ERROR: 'ADD_ERROR',
  SIGNING_UP: 'SIGNING_UP',
  SIGNUP_SUCCESS: 'SIGNUP_SUCCESS',
  SIGNUP_ERROR: 'SIGNUP_ERROR',
};
