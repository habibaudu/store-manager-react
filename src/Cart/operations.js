import axios from 'axios';
import { toast } from 'react-toastify';
import actions from './actions';
import constants from './constants';

const API_URL = process.env.API_URL;
const { setSalesRecord, setSalesError } = actions;
const url = `${API_URL}sales`;

const createSalesRecord = makesales => dispatch => {
  dispatch(setSalesRecord(constants.MAKING_SALES));
  const token = localStorage.getItem('token');
  const defaultOptions = {
    headers: {
      'x-access-token': token,
    },
  };
  return axios
    .post(
      'https://store-manager2.herokuapp.com/api/v1/sales',
      makesales,
      defaultOptions,
    )
    .then(({ data }) => {
      dispatch(setSalesRecord(constants.SALES_SUCESS));
    })
    .catch(({ response }) => {
      dispatch(setSalesError(constants.SALES_ERROR));
      toast.error(response, {
        hideProgressBar: true,
      });
    });
};

export { createSalesRecord };
