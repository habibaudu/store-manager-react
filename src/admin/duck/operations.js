import axios from 'axios';
import { toast } from 'react-toastify';
import actions from './actions';
import constants from './constants';

const API_URL = process.env.API_URL;
const {
  setAllProducts,
  addProducts,
  addProductStatus,
  addProductError,
  setDeleteProduct,
  setSignupError,
  setSignupState,
} = actions;

const url = `${API_URL}products`;

const fetchAllProducts = () => dispatch => {
  const token = localStorage.getItem('token');
  const defaultOptions = {
    headers: {
      'x-access-token': token,
    },
  };
  return axios
    .get('https://store-manager2.herokuapp.com/api/v1/products', defaultOptions)
    .then(({ data }) => {
      const img = data.imgArr;
      const data2 = data.rows;
      let count = 0;
      const products = [];
      data2.forEach(product => {
        const {
          id,
          productname,
          price,
          quantity,
          minimum,
          description,
          modified_date,
          created_date,
        } = product;

        products.push({
          item: img[count],
          id,
          productname,
          price,
          quantity,
          minimum,
          description,
          modified_date,
          created_date,
        });

        count += 1;
      });
      dispatch(setAllProducts(products));
    })
    .catch(({ response }) => {
      toast.error(response, {
        hideProgressBar: true,
      });
    });
};

export const addProduct = (
  productname,
  price,
  minimum,
  quantity,
  images,
  description,
) => dispatch => {
  dispatch(addProductStatus(constants.ADDING_PRODUCT));
  const token = localStorage.getItem('token');
  const defaultOptions = {
    headers: {
      'x-access-token': token,
    },
  };

  return axios
    .post(
      'https://store-manager2.herokuapp.com/api/v1/products',
      { productname, minimum, description, images, price, quantity },
      defaultOptions,
    )
    .then(({ data }) => {
      dispatch(addProductStatus(constants.PRODUCT_SUCESSFULLY_ADDED));
      dispatch(addProducts(data));
    })
    .catch(({ response }) => {
      toast.error(response, {
        hideProgressBar: true,
      });
      dispatch(addProductError(constants.ADD_ERROR));
    });
};

export const deleteProduct = id => dispatch => {
  const deleteUrl = `https://store-manager2.herokuapp.com/api/v1/products/${id}`;
  const token = localStorage.getItem('token');
  const defaultOptions = {
    headers: {
      'x-access-token': token,
    },
  };

  return axios
    .delete(deleteUrl, defaultOptions)
    .then(({ data }) => {
      dispatch(setDeleteProduct(data));
    })
    .catch(({ response }) => {
      toast.error(response, {
        hideProgressBar: true,
      });
    });
};

export const createAttendant = (
  email,
  password,
  username,
  Role,
) => dispatch => {
  const token = localStorage.getItem('token');
  const defaultOptions = {
    headers: {
      'x-access-token': token,
    },
  };
  dispatch(setSignupState(constants.SIGNING_UP));
  dispatch(setSignupError(''));
  return axios
    .post(
      `https://store-manager2.herokuapp.com/api/v1/auth/signup`,
      {
        email,
        password,
        username,
        Role,
      },
      defaultOptions,
    )
    .then(({ data }) => {
      dispatch(setSignupState(constants.SIGNUP_SUCCESS));
    })
    .catch(({ response }) => {
      dispatch(setSignupState(constants.SIGNUP_ERROR));
      dispatch(setSignupError(response));
      toast.error(response, {
        hideProgressBar: true,
      });
    });
};

export { fetchAllProducts };
