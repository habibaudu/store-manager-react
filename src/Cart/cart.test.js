import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import axios from 'axios';
import {
  CartComponent,
  CartContainer,
  mapDispatchToProps,
} from './CartComponent';
import cartReducer from './reducer';
import types from './types';
import constants from './constants';
import actions from './actions';
import { createSalesRecord } from './operations';

jest.mock('axios');

const { setSalesRecord, setSalesError } = actions;
const props = {
  addToCart: jest.fn(),
  removeFromCart: jest.fn(),
  salesRecord: jest.fn(),

  cart: {
    data: [
      {
        id: 2,
        attendants_id: 3,
        productname: 'shoe',
        price: 2345,
        quantity: 34,
      },
      {
        id: 22,
        attendants_id: 4,
        productname: 'shoe',
        price: 2000,
        quantity: 34,
      },
    ],
  },
};

describe('SALES ATTENDANT RECORD TEST SUITE', () => {
  describe('Record Component', () => {
    it('should render the cart Page', () => {
      const component = shallow(<CartComponent {...props} />);
      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    });

    it('should render the Cart container', () => {
      const component = shallow(<CartContainer />);
      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    });
  });

  it('should ensures mapDispatchToProps dispatches the specified actions', () => {
    const mock = configureStore([thunk]);
    const store = mock(props);
    const component = shallow(
      <Provider store={store}>
        <CartContainer />
      </Provider>,
    );
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).addToCart();
    mapDispatchToProps(dispatch).removeFromCart();
    mapDispatchToProps(dispatch).salesRecord();
  });
});

describe('TEST FOR CHECHOUT', () => {
  const payload = {
    data: {
      productname: 'productname',
      id: 1,
      quantity: 1,
      minimum: 2,
    },
  };
  const mockStore = configureStore([thunk]);
  const store = mockStore({});
  let wrapper;
  const response = { message: 'add to cart' };
  axios.post.mockImplementation(() =>
    Promise.resolve({ message: { ...response } }),
  );
  store.dispatch(createSalesRecord()).then(() => {
    store.dispatch('ADD_TO_CART');
  });
});

describe('cart Reducers', () => {
  it('should setup default state values', () => {
    const state = cartReducer(undefined, {
      type: '@@INIT',
    });
    expect(state).toEqual({ data: [] });
  });

  it('it should add one product to cart state', () => {
    const action = {
      type: 'ADD_TO_CART',
      data: [{ quantityInCart: 1 }],
    };
    const state = cartReducer(undefined, action);
    expect(state.data).toEqual(action.data);
  });

  it('it should change make sales error state', () => {
    const action = {
      type: 'MAKE_SALES_ERROR',
      salesError: undefined,
    };
    const state = cartReducer(undefined, action);
    expect(state.undefined).toEqual(action.undefined);
  });
});

describe('CART Actions', () => {
  it('it should set the sales record state', () => {
    const action = setSalesRecord(constants.SALES_SUCESS);
    expect(action).toEqual({
      type: types.MAKE_SALES,
      salesState: constants.SALES_SUCESS,
    });
  });

  it('it should set the sales error message', () => {
    const action = setSalesError(constants.SALES_ERROR);
    expect(action).toEqual({
      type: types.MAKE_SALES_ERROR,
      salesError: constants.SALES_ERROR,
    });
  });
});
