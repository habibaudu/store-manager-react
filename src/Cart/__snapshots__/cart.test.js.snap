// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SALES ATTENDANT RECORD TEST SUITE Record Component should render the Cart container 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`SALES ATTENDANT RECORD TEST SUITE Record Component should render the cart Page 1`] = `
<Fragment>
  <div>
    <div
      className="formbox"
    >
      <div
        className="log_head"
      >
        <h1>
          Your Cart
        </h1>
      </div>
      <div
        className="article"
        id="cartproduct"
      >
        <div
          key="shoe"
        >
          <div
            className="show"
          >
            <img
              alt=" "
              src=""
            />
          </div>
          <div
            className="discrip"
          >
            <span
              className="note"
            >
              shoe
            </span>
            <br />
            <span
              className="note"
            >
              Price :$
              2345
            </span>
            <br />
            <span
              className="note"
            >
              Quantity In Cart :
            </span>
            <br />
            <span
              className="note"
            >
              Quantity in Store :
              34
            </span>
            <br />
            <button
              className="bt2"
              onClick={[Function]}
            >
              Add
            </button>
            <button
              className="bt2"
              onClick={[Function]}
            >
              Remove
            </button>
            <br />
            <br />
            <br />
            <br />
          </div>
        </div>
        <div
          key="shoe"
        >
          <div
            className="show"
          >
            <img
              alt=" "
              src=""
            />
          </div>
          <div
            className="discrip"
          >
            <span
              className="note"
            >
              shoe
            </span>
            <br />
            <span
              className="note"
            >
              Price :$
              2000
            </span>
            <br />
            <span
              className="note"
            >
              Quantity In Cart :
            </span>
            <br />
            <span
              className="note"
            >
              Quantity in Store :
              34
            </span>
            <br />
            <button
              className="bt2"
              onClick={[Function]}
            >
              Add
            </button>
            <button
              className="bt2"
              onClick={[Function]}
            >
              Remove
            </button>
            <br />
            <br />
            <br />
            <br />
          </div>
        </div>
        <span
          className="note"
        >
          Total :
          NaN
        </span>
        <br />
        <form
          id="cart"
          onSubmit={[Function]}
        >
          <input
            type="submit"
            value="Checkout"
          />
          <br />
          <br />
          <br />
          <br />
        </form>
      </div>
      <div
        className="form_footer clearfix"
      >
        <p />
      </div>
    </div>
  </div>
</Fragment>
`;
